components/org.wso2.carbon.identity.multi.attribute.login/org.wso2.carbon.identity.multi.attribute.login.resolver.regex/src/main/java/org/wso2/carbon/identity/multi/attribute/login/resolver/regex/RegexResolver.java

```diff
@@ -18,91 +18,47 @@
 
 package org.wso2.carbon.identity.multi.attribute.login.resolver.regex;
 
-import org.apache.commons.lang.StringUtils;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.login.resolver.regex.RegexLoginResolver;
 import org.wso2.carbon.identity.multi.attribute.login.mgt.MultiAttributeLoginResolver;
 import org.wso2.carbon.identity.multi.attribute.login.mgt.ResolvedUserResult;
-import org.wso2.carbon.identity.multi.attribute.login.resolver.regex.utils.UserResolverUtil;
-import org.wso2.carbon.user.api.Claim;
-import org.wso2.carbon.user.api.ClaimManager;
-import org.wso2.carbon.user.api.UserStoreException;
-import org.wso2.carbon.user.core.UniqueIDUserStoreManager;
-import org.wso2.carbon.user.core.UserRealm;
 import org.wso2.carbon.user.core.common.AuthenticationResult;
-import org.wso2.carbon.user.core.common.User;
-import org.wso2.carbon.user.core.constants.UserCoreClaimConstants;
-import org.wso2.carbon.user.core.util.UserCoreUtil;
 
 import java.util.List;
-import java.util.regex.Pattern;
 
 /**
  * This class is used to implement MultiAttributeLoginResolver. In this class, users will be resolved using a regex
  * pattern.
+ *
+ * @deprecated To generalize the resolver concept and make it extensible.
+ * Use the {@link org.wso2.carbon.identity.login.resolver.regex.RegexLoginResolver} class instead.
  */
+@Deprecated
 public class RegexResolver implements MultiAttributeLoginResolver {
 
+    private static final RegexLoginResolver regexLoginResolver = new RegexLoginResolver();
```

Comment:
As this is static multiple threads will share the same instance. We also have " private static final Log log = LogFactory.getLog(RegexLoginResolver.class);" in the RegexLoginResolver class. So this will be shared between threads when logging. However it is highly unlikly to be a Logger implementation to not to be thread safe. Even it is thread safe; there can be a perf issue in case of a error continiously logging from this class obj. and multiple threads are sharing the same resource. but this is highly unlikly too.

Reply:
So better to avoid unnecessary loggin